resteasy-jaxrs/src/main/java/org/jboss/resteasy/microprofile/config/ResteasyConfig.java

```diff
@@ -102,20 +137,38 @@ public String getValue0(String propertyName, SOURCE source, String defaultValue)
       }
       else
       {
-         return config.getOptionalValue(propertyName, String.class).orElse(defaultValue);
+         try
+         {
+            final Object config = GET_CONFIG.invoke(null, getClassLoader());
+            Optional<String> opt = (Optional<String>) GET_OPTIONAL_VALUE.invoke(config, propertyName, String.class);
+            return opt.orElse(defaultValue);
+         }
+         catch (IllegalAccessException | InvocationTargetException e) {
+            LogMessages.LOGGER.debugf(e, "Failed to invoke the configuration API method %s.", GET_VALUE);
+            return defaultValue;
+        }
       }
    }
 
    public Iterable<String> getPropertyNames()
    {
-      return config == null ? null : getPropertyNames0();
+      return !MP_CONFIG_PRESENT ? null : getPropertyNames0();
    }
 
+   @SuppressWarnings("unchecked")
    public Iterable<String> getPropertyNames0()
    {
       if (System.getSecurityManager() == null)
       {
-         return config.getPropertyNames();
+         try
+         {
+            final Object config = GET_CONFIG.invoke(null, getClassLoader());
+            return (Iterable<String>) GET_PROPERTY_NAMES.invoke(config);
+         }
+         catch (IllegalAccessException | InvocationTargetException e) {
+            LogMessages.LOGGER.debugf(e, "Failed to invoke the configuration API method %s.", GET_VALUE);
```

Comment:
The argument should be `GET_PROPERTY_NAMES`.

Reply:
Same ...